tests/unittests/utils/test_ptv3.py

```
@@ -0,0 +1,160 @@
+
+    def test_pads_missing_intensity_with_zeros(self):
+        model = _make_model()
+        cloud = np.random.rand(50, 3).astype("float32")  # xyz only
+        point = model._build_input(cloud)
+
+        assert point["feat"].shape[1] == 4
+        assert torch.count_nonzero(point["feat"][:, 3]) == 0
+
+    def test_rejects_too_few_columns(self):
+        model = _make_model()
+        with pytest.raises(ValueError):
```

Ruff (0.15.17)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)